rpc/monero-rpc-types/src/json.rs

```diff
@@ -0,0 +1,81 @@
+//! JSON types.
+//!
+//! <https://github.com/monero-project/monero/blob/cc73fe71162d564ffda8e549b79a350bca53c454/src/rpc/daemon_messages.h>.
+
+//---------------------------------------------------------------------------------------------------- Import
+use crate::macros::define_monero_rpc_struct;
+
+//---------------------------------------------------------------------------------------------------- Struct definitions
+define_monero_rpc_struct! {
+    // The markdown tag for Monero RPC documentation. Not necessarily the endpoint.
+    get_block_count,
+    // The `$file.$extension` in which this type is defined in the Monero
+    // codebase in the `rpc/` directory, followed by the specific lines.
+    core_rpc_server_commands_defs.h => 919..=933,
+    // The actual type definitions.
+    // If there are any additional attributes (`/// docs` or `#[derive]`s)
+    // for the struct, they go here, e.g.:
+    // #[derive(MyCustomDerive)]
+    GetBlockCount, // <- The type name.
+    #[derive(Copy)]
+    Request /* <- The request type */ {
+        // This request type requires no inputs,
+        // so it is left empty.
+    },
+    #[derive(Copy)]
+    Response /* <- The response type */ {
+        // Within the `{}` is an infinite matching pattern of:
+        // ```
+        // $ATTRIBUTES
+        // $FIELD_NAME: $FIELD_TYPE,
+        // ```
+        // The struct generated and all fields are `pub`.
+
+        count: u64,
+        status: crate::Status,
+        untrusted: bool,
+    }
+}
+
+define_monero_rpc_struct! {
+    on_get_block_hash,
+    core_rpc_server_commands_defs.h => 935..=939,
+    OnGetBlockHash,
+    #[derive(Copy)]
+    Request {
+        block_height: u64,
+    },
+    Response {
+        block_hash: String,
+    }
+}
```

Comment:
Another smaller example, this defines `crate::json::OnGetBlockHash{Request,Response}` which gets re-exported as `crate::{req,resp}::json::GetBlockCount`.